CDCgov/NEDSS-Modernization
# [CNFT1-1559] fixes gender searches for Patient Search

## Description

Changes how gender affects Patient Search by being more lenient with the accepted values and including Patients without a specified gender when searching for `Other` genders.

- `PatientFilter` no longer uses the `Gender` enum
- Gender searches for `Other` now include Patients with a gender of `Unknown` or no specified gender.
- Refactors the creation of the query used to search patients in Elasticsearch to resolve the "high cognitive complexity" Sonar find.

## Tickets

* [CNFT1-1559](https://cdc-nbs.atlassian.net/browse/CNFT1-1559)

## Steps to verify

1. Navigate to Advanced Search and search for Patients by Gender either `Female` or `Male`.
2. Click `Clear all` and attempt to search again
3. The `You must enter at least one item to search` should appear
4. Search for Patients with a gender of `Other`
5. The results should include Patients with no specified gender as well as Patients with an `Unknown` gender.


[CNFT1-1559]: https://cdc-nbs.atlassian.net/browse/CNFT1-1559?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
